### DISCUSSION TITLE
upload existing theme that was directly made in my own editor

### DISCUSSION THREAD

**Discussion by artivilla** [labithiotis/zed-themes#113] [Ideas]
Great project!🥳🥳

I just released a new fairly popular theme called [Ariake](https://github.com/artivilla/zed-ariake-theme) but it was directly created in my editor. Can we include an import theme to upload json files directly to this interface.

**Comment by labithiotis** [comment]
@artivilla If you click on the caret icon on the create button you have the option to upload JSON theme, you can then sign in and publish the theme. Would that suit your needs?
![image](https://github.com/user-attachments/assets/67cbaab3-4e6a-4ea5-aee9-0a04602cd431)

**Comment by artivilla** [comment]
Unclear why it didn't upload. The theme loads fine locally.

![image](https://github.com/user-attachments/assets/049a6f33-b273-4de7-966a-e7b4132b2db7)

![image](https://github.com/user-attachments/assets/1a671a13-1a27-489a-b5a3-86833a785ffb)

![image](https://github.com/user-attachments/assets/a67dd7ab-a4f7-44a2-9376-1803f50b0edd)

**Comment by labithiotis** [comment]
@artivilla Darn, I do need to see about implementing a nicer error handler, I just use alert as a quick win for now.

As for the error, seems to be saying the JSON is not valid, does the JSON pass validation on https://jsonlint.com ?

**Reply by artivilla** [reply]
all good. your customers are mostly devs so they can help themselves.

**Comment by artivilla** [comment]
Nvm. works! I think the theme builder is more lenient when you override it in `~/.config/zed/settings.json` and accepts comments in json vs when you port it over to your actual theme. Ty! will be submitting my themes soon. Although might be worth automating and pulling them from the [official store ](https://github.com/zed-industries/extensions/tree/main/extensions) as well.

**Reply by labithiotis** [reply]
@artivilla Glad you resolved it, I do already pull any official themes from https://github.com/zed-industries/extensions regularly, it last ran 4 hours ago https://github.com/labithiotis/zed-themes/actions/runs/11361406690/job/31601044179#step:7:1 but I don't your theme in that repo?

**Reply by artivilla** [reply]
No, but I'm actually submitting it as we speak so maybe I don't need to put it here :) 
Might be worth mentioning that some where, likely at the point of upload. Still really nice work to be able to quickly view themes.

**Reply by labithiotis** [reply]
That's a good point, I should make it much clearer that themes added to the official repo will auto-sync here. Did think I should include links back to repos and collect star counts etc...

**Reply by artivilla** [reply]
Didn't understand the last part? Yeah worth asking authors to include links to your repository. I think the best part is being able to edit it live and a killer feature would be `Highlight each part of the IDE as you hover over token?` as you mentioned in your readme. I think this is where zed falls apart and the value add of your product.

Authors can still include screenshots in their repos and its easy to try them in the editor itself so viewing is secondary imo. Its very hard to 'create/edit' them however.

**Reply by labithiotis** [reply]
I merely meant I should include additional information in the same way Zed does, like downloads/stars/repo link etc.. 
![image](https://github.com/user-attachments/assets/d93236e1-29aa-4128-a89e-23fd8f6a94d5)

Highlighting is also on my todo list and others have requested it, as well as many others, but this is just a side project I do in my spare time :)

**Reply by artivilla** [reply]
yeah totally! yes, stats would be interesting for sure.

**Comment by eproxus** [comment]
I can't see any upload button. Did it disappear?

**Comment by labithiotis** [comment]
@eproxus The button to upload a theme should still be there on the home page?

![image](https://github.com/user-attachments/assets/7fa6cc12-cfcd-42d5-84d4-79ebadd29d6d)

**Reply by eproxus** [reply]
Ah, I was looking for it in the editor. Didn't know about the front page.
